pkg/extensions/sync/service.go

```diff
@@ -229,12 +238,42 @@ func (service *BaseService) GetNextRepo(lastRepo string) (string, error) {
 			break
 		}
 
+		if service.clusterConfig != nil {
+			targetIdx, targetMember := computeTargetMember(service.clusterConfig, lastRepo)
+
+			// if the target index does not match with the local socket index,
+			// then the local instance is not responsible for syncing the repo and should skip the sync
+			if targetIdx != service.clusterConfig.Proxy.LocalMemberClusterSocketIndex {
+				service.log.Debug().
+					Str(constants.RepositoryLogKey, lastRepo).
+					Str("targetMemberIndex", fmt.Sprintf("%d", targetIdx)).
+					Str("targetMember", targetMember).
+					Msg("skipping sync of repo not managed by local instance")
+
+				continue
+			}
+		}
+
 		matches = service.contentManager.MatchesContent(lastRepo)
 	}
 
 	return lastRepo, nil
 }
 
+// computes the target member using siphash and returns the index and the member
+// siphash was chosen to prevent against hash attacks where an attacker
+// can target all requests to one given instance instead of balancing across the cluster
+// resulting in a Denial-of-Service (DOS).
+// ref: https://en.wikipedia.org/wiki/SipHash
+func computeTargetMember(clusterConfig *config.ClusterConfig, name string) (uint64, string) {
```

Comment:
This logic is common to `proxy.go` as well as `service.go` which are in different packages. However, it is not common enough to be placed in the `common` package.

@rchincha do you know which package might be appropriate to have this shared hash compute logic?

Reply:
This is tricky since sync is an extension which can be optionally compiled. So maybe put it in cluster pkg and make it available to sync?

Reply:
I moved the code to a new cluster package and generalized the parameters so that it can be used in both `api` and `sync` packages